src/io/fabric8/Fabric8Commands.groovy

```diff
@@ -150,23 +150,50 @@ def getNewVersionFromTag(){
     return version
   }
 
-  tag = tag.trim()
-
-  // strip the v prefix from the tag so we can use in a maven version number
-  def previousReleaseVersion = tag.substring(tag.lastIndexOf('v')+1)
-  echo "Previous version found ${previousReleaseVersion}"
+  def semver = tag =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+
+  if (semver.matches()) {
+    def majorVersion = semver.group('major') as int
+    def minorVersion = (semver.group('minor') ?: 0) as int
+    def patchVersion = ((semver.group('patch') ?: 0) as int) + 1
+
+    def pomSemver = pomVersion =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+    if (pomSemver.matches) {
+      def pomMajorVersion = pomSemver.group('major') as int
+      def pomMinorVersion = (pomSemver.group('minor') ?: 0) as int
+      def pomPatchVersion = (pomSemver.group('patch') ?: 0) as int
+
+      if (pomMajorVersion > majorVersion) {
+        majorVersion = pomMajorVersion
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion > minorVersion) {
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion == minorVersion && pomPatchVersion > patchVersion) {
+        patchVersion = pomPatchVersion
+      }
```

Comment:
I have mixed feelings, too. For one you can do still diffs between tags, thats good. But you can't do stuff like 'git log v3.1.3:v3.1.4' to get all log comments between two releases. That's bad (but maybe there is another solution, my git fu is limited). And in visual tools (my favourite is still 'gitup') you wont see any tag labels on the branches. Just a commit number, if any.

So it *has* advantages to stick tags to branches. 

The drawback is of course the PR and merge scenario. The only big problem I see here is, that while the test is running on the release branch someone pushes a non-mergable change to master in the meantime. Thats probably not easily resolvable, too.

But what about to *not* delete the branch and giving it a meaningful name (like 'branch-v3.1.4') ? Then you can least see the branch-off points in your history (not sure about the logs, though).